src/main/java/com/tradeshift/amqp/autoconfigure/TunedRabbitAutoConfiguration.java

```diff
@@ -67,11 +65,14 @@
 
     private final ApplicationContext applicationContext;
     private final ConfigurableListableBeanFactory beanFactory;
+	private final RabbitComponentsFactory rabbitComponentsFactory;
```

Comment:
Please, remove this beans as a dependancy. We need to create this guy here to respect the creation flow of the beans.
```suggestion
@Bean
    public RabbitComponentFactory rabbitComponentFactory() {
        return new RabbitComponentsFactory();
    }
```

Reply:
This [test](https://github.com/wesleyegberto/spring-rabbitmq-tuning/blob/20e8a497b4442f6b1f3637c51f5ba83c50f7fcd1/src/test/java/com/tradeshift/amqp/autoconfigure/TunedRabbitAutoConfigurationTest.java#L520) is using this dependency to allow it to mock the RabbitAdmin and check if the exchange and queue creation was called correctly.

Removing it I think we will need to change [this](https://github.com/wesleyegberto/spring-rabbitmq-tuning/blob/20e8a497b4442f6b1f3637c51f5ba83c50f7fcd1/src/main/java/com/tradeshift/amqp/autoconfigure/TunedRabbitAutoConfiguration.java#L168) method to receive the RabbitComponentsFactory allowing the test to pass the mocked instance.
Or we could create the bindings tests in another test class using `@Spy` at TunedRabbitAutoConfiguration to allow it to mock the method `rabbitComponentFactory()`.

What do you think about it? Maybe there is a better way that I'm missing.

I feel that the class `TunedRabbitAutoConfiguration` and its method`routingConnectionFactory` is doing too much.
Maybe extract another class with the logic to control the creation of ConnectionFactory (by Host+Port and VirtualHost) and keep the class with only beans declarations.

Reply:
Hmm, let's solve one problem per time.

To solve the bean problem the only think that you need to do is use the same strategy that you are using to the RabbitAdmin.

`RabbitComponentFactory rabbitComponentFactory = mock(RabbitComponentFactory.class);`

Reply:
About the responsibility of `routingConnectionFactory`. I think we can follow with the current implementation and after this we can open an issue to split it.

What you think?

Reply:
Done. Could you check the changes?